internal/http/middleware/authorization/kessel.go

```diff
@@ -0,0 +1,162 @@
+	"google.golang.org/grpc"
+)
+
+func NewKesselClient(config config.Config) KesselMiddlewareBuilder {
+	options := []func(*common.Config){
+		common.WithgRPCUrl(config.KesselURL),
+		common.WithTLSInsecure(config.KesselInsecure),
+	}
+
+	if config.KesselAuthEnabled {
+		options = append(options, common.WithAuthEnabled(config.KesselAuthClientID, config.KesselAuthClientSecret, config.KesselAuthOIDCIssuer))
+	}
+
+	client, _ := v1beta2.New(common.NewConfig(options...))
+
+	return &kesselMiddlewareBuilderImpl{
+		client:                   client,
+		config:                   config,
+		defaultWorkspaceResolver: getDefaultWorkspaceID,
+	}
+}
+
+type KesselMiddlewareBuilder interface {
+	EnforceDefaultWorkspacePermission(permission string) func(http.Handler) http.Handler
+	EnforceDefaultWorkspacePermissionForUpdate(permission string) func(http.Handler) http.Handler
+}
+
+type defaultWorkspaceResolver func(context.Context, string) (string, error)
+
+type kesselMiddlewareBuilderImpl struct {
+	client                   *v1beta2.InventoryClient
+	config                   config.Config
+	defaultWorkspaceResolver defaultWorkspaceResolver
+}
+
+var _ KesselMiddlewareBuilder = &kesselMiddlewareBuilderImpl{}
+
+type AllowedResponse interface {
+	GetAllowed() kesselv2.Allowed
+}
+
+type kesselCheckFn func(context.Context, *kesselv2.ResourceReference, string, *kesselv2.SubjectReference, ...grpc.CallOption) (AllowedResponse, error)
+
+func (a *kesselMiddlewareBuilderImpl) callCheck(ctx context.Context, object *kesselv2.ResourceReference, relation string, subject *kesselv2.SubjectReference, opts ...grpc.CallOption) (AllowedResponse, error) {
+	request := &kesselv2.CheckRequest{
+		Object:   object,
+		Relation: relation,
+		Subject:  subject,
+	}
+
+	return a.client.KesselInventoryService.Check(ctx, request, opts...)
+}
+
+func (a *kesselMiddlewareBuilderImpl) callCheckForUpdate(ctx context.Context, object *kesselv2.ResourceReference, relation string, subject *kesselv2.SubjectReference, opts ...grpc.CallOption) (AllowedResponse, error) {
+	request := &kesselv2.CheckForUpdateRequest{
+		Object:   object,
+		Relation: relation,
+		Subject:  subject,
+	}
+
+	return a.client.KesselInventoryService.CheckForUpdate(ctx, request, opts...)
+}
+
+func (a *kesselMiddlewareBuilderImpl) EnforceDefaultWorkspacePermission(permission string) func(http.Handler) http.Handler {
+	return a.enforceOrgPermission(permission, a.callCheck)
+}
+
+func (a *kesselMiddlewareBuilderImpl) EnforceDefaultWorkspacePermissionForUpdate(permission string) func(http.Handler) http.Handler {
+	return a.enforceOrgPermission(permission, a.callCheckForUpdate)
+}
+
+func (a *kesselMiddlewareBuilderImpl) enforceOrgPermission(permission string, checkFn kesselCheckFn) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if !a.config.KesselEnabled {
+				next.ServeHTTP(w, r)
+				return
```

Comment:
This does mean we don't fallback to RBAC? and POC is about Kessel and not supporting the v1 or RBAC? is this correct?

Reply:
There has never been an RBACv1 implementation in config manager, hence there is nothing to fall back to. If Kessel is disabled, the current behavior (no authorization enforcement) will be achieved.